tests/test_NSArray.py

```diff
@@ -276,13 +276,13 @@ def initWithList_(self, data):
 
         # If it's set through a method call, it becomes an objc instance
         obj2 = PrimitiveListAttrContainer.alloc().initWithList_([4, 5, 6])
-        self.assertEqual(obj2.data, [4, 5, 6])
         self.assertIsInstance(obj2.data, ObjCListInstance)
+        self.assertEqual(py_from_ns(obj2.data), [4, 5, 6])
```

Comment:
That's because this code is actually dealing with `NSNumber`s, it's just not very obvious :) When a Python object is passed into an Objective-C object parameter (or stored in an Objective-C object property), it's [automatically converted using `ns_from_py`/`at`](https://rubicon-objc.readthedocs.io/en/latest/how-to/type-mapping.html#converting-from-python-to-objective-c) to an equivalent Objective-C data structure. In this case that means that the list of ints is being converted to a `NSArray` of `NSNumber`s when it's passed into `initWithList_` or assigned to the `data` property.

Previously the reverse conversion (`NSNumber` to Python `int`) would happen automatically - the array's elements are internally retrieved using `array.objectAtIndex(i)`, so they would be unwrapped by the automatic `NSNumber` conversion. The Objective-C method calls are hidden by our Python-style `ObjCListInstance` interface, so it's less obvious that the conversion previously applied here.

I'll add an extra paragraph to the change notes to point this out.